src/ORM/Association.php

```diff
@@ -438,6 +438,15 @@ public function property($name = null)
         if ($name === null && !$this->_propertyName) {
             list(, $name) = pluginSplit($this->_name);
             $this->_propertyName = Inflector::underscore($name);
+            $columns = array_keys($this->_sourceTable->schema()->columns());
```

Comment:
No need to call `array_keys()`. [Cake\Database\Schema\Table::columns()](https://github.com/cakephp/cakephp/blob/2a1d9810714ab48bf2877c291ef0063763be3b8b/src/Database/Schema/Table.php#L318) seem to return a non-assoc array.